### ISSUE TITLE
 All players have the same IP - TCPShield workaround version (192.168.0.1)

### ISSUE THREAD

**Issue by PolsatGraniePL** [games647/FastLogin#728]
### What happened?

 All players have the same IP.  192.168.0.1

### Server log

```shell
Spigot server:

[09:57:32 INFO]: PolsatGraniePL[/192.168.0.1:57242] logged in with entity id 95 at ([world]9.5, 98.0, -4.5)
[09:57:32 WARN]: [Essentials] Permissions lag notice with (LuckPermsHandler). Response took 83.455420ms. Summary: Getting prefix for PolsatGraniePL
[09:57:33 INFO]: [FastLogin] No on-going login session for player: CraftPlayer{name=PolsatGraniePL} with ID 54.XX.XXX.XXX:57242 (Server/VPS IP)
[09:57:50 INFO]: PolsatGraniePL issued server command: /captcha ufh69
[09:57:56 INFO]: [AuthMe] PolsatGraniePL logged in 192.168.0.1
[09:57:57 INFO]: [AuthMe] PolsatGraniePL registered 192.168.0.1


velocity server:

[09:57:31 INFO] [fastlogin]: Incoming login request for PolsatGraniePL from /31.XX.XXX.XXX:17787 (My IP)
[09:57:31 INFO] [fastlogin]: Handling player PolsatGraniePL
[09:57:31 INFO]: [connected player] PolsatGraniePL (/31.XX.XXX.XXX:17787 (My IP)) has connected
[09:57:32 INFO]: [server connection] PolsatGraniePL -> hub has connected
```

### Plugin version

1.11-SNAPSHOT-d4a424f

### Platform

Velocity

### Relevance

- [X] I tried the latest build
- [x] I checked for existing tickets -
If there are, please vote them with a thumps reaction and not create new ones

**Comment by games647** [comment]
That's 192 is a local ip address. Check your proxy, floodgate etc settings. FastLogin only reads the IP and doesn't change it.

**Comment by PolsatGraniePL** [comment]
only that when I remove the fastlogin plugin, it then works normally.
![image](https://user-images.githubusercontent.com/88681446/154031148-9a7198e6-92fa-4b4b-8fa5-e65be5fc2009.png)



And when the plugin is installed, it changes the IP
```
[09:57:32 INFO]: PolsatGraniePL[/192.168.0.1:57242] logged in with entity id 95 at ([world]9.5, 98.0, -4.5)
[09:57:32 WARN]: [Essentials] Permissions lag notice with (LuckPermsHandler). Response took 83.455420ms. Summary: Getting prefix for PolsatGraniePL
[09:57:33 INFO]: [FastLogin] No on-going login session for player: CraftPlayer{name=PolsatGraniePL} with ID 54.XX.XXX.XXX:57242 (Server/VPS IP)
[09:57:50 INFO]: PolsatGraniePL issued server command: /captcha ufh69
[09:57:56 INFO]: [AuthMe] PolsatGraniePL logged in 192.168.0.1
[09:57:57 INFO]: [AuthMe] PolsatGraniePL registered 192.168.0.1
```

**Comment by games647** [comment]
It doesn't. `192.168.0.1` is a local IP address. As you can see the plugin also reads the correct IP address later on. Floodgate  (or any other plugin like TCPShield) likely modifies the IP address, but cannot find the player again after the UUID of the player got changed by FastLogin. So please post the plugin list and test the plugins individually.

**Comment by games647** [comment]
BTW: I saw that you use Velocity. Check if IP forwarding correctly configured.

**Comment by From-pErfo** [comment]
I had this problem too, using that solution [But if you don't use it, quick login won't work]
Refers to other plugins can not get the correct ip[becomes 192.168.x]
I still use proxy )
```
[FastLogin] Handling player From_pErfo
[FastLogin] Requesting premium login for registered player: From_pErfo
[FastLogin] Enabling onlinemode encryption for /[IPv6]:60304
[FastLogin] Profile From_pErfo has a verified premium account
From_pErfo[/192.168.0.1:60304] logged in with entity id 135831 at ([world]0, 0, 0)
```

**Comment by arima0k** [comment]
check your router NAT config

**Comment by CroaBeast** [comment]
I have the same issue in my spigot server (without bungeecord).
I disabled FastLogin and all the IP's are correct.
Is there any way to prove this issue?

**Comment by games647** [comment]
@CroaBeast FastLogin doesn't change the IP of connections. `192.168.0.1` is the a local IP address. This means you have some form of proxy that isn't correctly forwarding the real IP address if FastLogin is installed. Please post your plugins list. One example is TCPShield for causing this..

**Comment by HuJK** [comment]
I think it's FastLogin's bug. My internal network uses 172.22.3.2, gateway is 172.22.0.1. But the logib IP still be 192.168.0.1.
FastLogin removed:
![image](https://user-images.githubusercontent.com/15907444/162576161-b7d24752-0be3-4bbc-bed7-c2d583dbfb26.png)

FastLogin installed:
![image](https://user-images.githubusercontent.com/15907444/162576315-8e152080-896e-4d5e-a2db-911e7d9971df.png)

**Comment by CroaBeast** [comment]
Without FastLogin, already disabled to avoid IP issues.
Also it's very weird that at least 3 people report the same issue besides me.
```
Plugins (46): AuthMe, BetterConcrete, Celeste, Citizens, CommandPanels, CommandRegions*, CommandWhitelist, ConditionalEvents, ConsoleSpamFix, CoreProtect, Crawling, DiscordSRV, FarmControl, InteractionVisualizer, ItemEdit, ItemTag, JakesRTP, LightAntiCheat, LightAPI, LiteBans, LuckPerms, MessageEditor, Multiverse-Core, Multiverse-Inventories, Multiverse-NetherPortals, MyCommand, PlaceholderAPI, ProtocolLib, RHLeafDecay, SIR*, SkinsRestorer, SlimeChunkChecker, spark, TAB, Vault, VeinMiner, ViaBackwards, ViaRewind, ViaVersion, VoidGen, VoidSpawn, WorldBorder, WorldEdit, WorldGuard, WorldGuardGUI, YamipaPlugin
```

**Comment by HuJK** [comment]
@games647 How to reproduce this:
https://drive.google.com/file/d/1qf6EpAKdLfYQgDnwm024BYSLeDgHaMW6/view?usp=sharing

1. run command
```
cd mc_slimefun_118_2/mc_server
./runserver_loop.sh
```
2. Login to the server
![image](https://user-images.githubusercontent.com/15907444/162582134-7cd4c364-2ae3-429f-88bc-f7f06e7a966a.png)

**Comment by games647** [comment]
@HuJK 

>I think it's FastLogin's bug

Doesn't have to be, it's only a correlation. First lets get the smallest reproducible configuration.

@HuJK @CroaBeast Please try it out with only auth plugin and FastLogin installed. There is a lot of chaos about what exactly the server configurations are. It's very important which server software you are using and what plugins are installed. Including if Minecraft native proxies are involved, see below. That's why I'm splitting up the tickets for proxy and without proxy setups.

There are big differences between a Minecraft proxy setup like the OP and ProtocolLib, ProtocolSupport spigot only handling. With Minecraft proxies the integration is not so deep. The plugin uses the direct BungeeCord/Velocity API onlinemode changes. The forwarding of details to Spigot then is done by BungeeCord and is the same like normal online mode. The Spigot plugin of FastLogin only gets the command to authenticate to the auth plugin. There are no changes to Minecraft internals here.

**Comment by HuJK** [comment]
Oh, I forgot to change the permission. Now it should be able to download.  
There are only three mod installed in the pack: FastLogin, AuthMe and protocollib.

**Comment by PolsatGraniePL** [comment]
ja swojego błędu nadal nie rozwiązałem, dlatego zrezygnowałem z użycia fastlogin.

**Comment by From-pErfo** [comment]
I get the correct IP address without using any proxy and proxy software
192.168.0.1 doesn't appear since I updated the version. but since I can't use proxy plugins, the IP address becomes 127.0.0.1 [this has nothing to do with any plugins]

In my impression, 192.168.0.1 only appears in the tcpshield_update version, I can't reproduce it in the new version anymore. Logs like this should be useless.

*my network environment only allows me to use a proxy.  Not having an IP is a serious security issue for me (whisper*
?

**Comment by HuJK** [comment]
This buggy version was downloaded from here: https://github.com/games647/FastLogin/releases

Problem resolved (for me) by upgrade to the latest version(Build 1070) of FastLogin download from https://ci.codemc.io/job/Games647/job/FastLogin/

I thought the version I downloaded from github release is the latest version bebore, maybe we should update it.

**Comment by games647** [comment]
@HuJK made me realize that this bug was introduced in d4a424f and fixed 4f3c8adc27adffe6723b48779fc82e8b991dc55b. Note this commit only belongs to separate branch. A special version only made for plugins like TCPShield or HAProxyDetector. It's different from the main development and was only published in form of GitHub releases with the note: `Temporarily fix for TCPShield.`

I'm sorry for the confusion. I wasn't aware that this separation isn't clear from a non-developer perspective. I pulled of the old version of that version and included a description for the newer one. That newer version already contains the fix. Users that don't use TCPShield or similar should download the normal version from Jenkins.

Details:
So that there is debug code in it (See [code details](https://github.com/games647/FastLogin/commit/d4a424f29717d36ed3b17f7f3f3f0160f73107b5#diff-95d121963a8f1e2415045bb68eb0db3f6259cf9d0850bd2f381a9d80bda3196eR110-R118)). This was fixed months ago in https://github.com/games647/FastLogin/commit/4f3c8adc27adffe6723b48779fc82e8b991dc55b.
